### ISSUE TITLE
FTPClient.connectionLost() doesn't fire errors on outstanding deferreds

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#8059]
|<img alt="fdrake's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| fdrake reported|
|-|-|
|Trac ID|trac#444|
|Type|defect|
|Created|2003-12-23 02:16:15Z|

<details><summary>Searchable metadata</summary>

```
trac-id__444 444
type__defect defect
reporter__fdrake fdrake
priority__high high
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__ftp ftp
keywords__ 
time__1072145775000000 1072145775000000
changetime__1469772013572894 1469772013572894
version__ 
owner__ 
cc__spiv cc__slyphon cc__fdrake
```
</details>

**Comment by twisted-trac** [comment]
|<img alt="fdrake's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|fdrake commented|
|-|-|

```
#!html
<pre>
When a server drops a connection to the client, FTPClient doesn't not fire the
errback for outstanding deferreds.  Extending the connectionLost() method of the
base class to call the fail() method ensures that deferreds are properly
notified of the dropped connection.
</pre>
```

**Comment by twisted-trac** [comment]
|<img alt="spiv's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|spiv commented|
|-|-|

```
#!html
<pre>
I'm assigning this to me, seeing as I'm the FTPClient expert, but I won't have
any time for this until after Christmas :(

This one sounds easily fixable, though, and a test case shouldn't be too hard
either.  If I haven't done this by the new year, slap me :)

(If I keep taking FTP bugs, I guess I should update ftp.py's maintainer line to
list me as well as slyphon...)
</pre>
```

**Comment by twisted-trac** [comment]
|<img alt="spiv's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|spiv commented|
|-|-|

```
#!html
<pre>
Well, I suck.  I finally (in March!) got around to fixing this.  

In my defence, it *was* a little harder than it looked... but not that much
harder :)
</pre>
```

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|<img alt="GitHub's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_5"></a>|GitHub commented|
|-|-|

In changeset dbd8cdb2ed148fb87839b360c7b1391a6c189a5f
```
#!CommitTicketReference repository="" revision="dbd8cdb2ed148fb87839b360c7b1391a6c189a5f"
Merge pull request #444 from twisted/8709-rodrigc-conch-recvline-py3

Author: rodrigc
Reviewer:  acabhishek942
Fixes: #8709

Fix twisted.conch.test.test_recvline on Python 3
```
